src/__tests__/utils.test.ts

```diff
@@ -117,3 +117,25 @@ describe('URI()', () => {
     });
   });
 });
+
+describe('maybeResolveLocalRef()', () => {
+  it('handles invalid $refs', () => {
+    expect(maybeResolveLocalRef({ a: true }, { $ref: '#a' })).toStrictEqual({
+      $ref: '#a',
```

Comment:
@XVincentX is this okay from Prism perspective or not necessarily?

Reply:
My point will always be the same: this should halt the parsing of this specific section and send an error up to the caller.

This will effectively let people feed Prism with an invalid document which will lead to an error on its own.

Reply:
> My point will always be the same: this should halt the parsing of this specific section and send an error up to the caller.

Yeah, I know your point of view and totally understand it.
This is not okay from the perspective of Studio, unfortunately. :disappointed: 

Would returning null/undefined be better here?

Reply:
None of these will make me happy — but with null we'd be at least ignoring the whole thing for real and that's better than leaving the reference unresolved. I'd vote for that.

Reply:
> None of these will make me happy 

Yeah, totally aware of it, that only errors will satisfy Prism, but it is what it is :man_shrugging: 
To be frank, I'm really leaning towards diverging http-spec, not sure whether it makes share the codebase.
The Studio and Prism needs vary a lot, and sharing the same package makes everything only more complicated for each tool respectively.

Speaking of code - will return null instead then.

Reply:
> To be frank, I'm really leaning towards diverging http-spec, not sure whether it makes share the codebase.

To me it does not. I've proposed that in the past and I received a no a response. Maybe if we join the forces, we can win this time?

Reply:
> Maybe if we join the forces, we can win this time?

:crossed_fingers: